docs/datadog_generic_resource.md

```diff
@@ -0,0 +1,173 @@
+# Datadog GenericResource
+
+This feature was introduced in Datadog Operator v1.12.0 and is currently in Preview.
+
+## Overview
+
+The `DatadogGenericResource` (DDGR) Custom Resource Definition allows users to create a variety of [Datadog resources](#supported-resources) using the Operator and manage them as Kubernetes resources. 
+
+Example:
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogGenericResource
+metadata:
+  name: browser-test-example
+spec:
+  type: synthetics_browser_test
+  jsonSpec: |-
+    {
+      "config": {
+        [...]
+      },
+      "locations": [
+        "aws:us-east-2"
+      ],
+      "message": "lorem ipsum",
+      "name": "Example Browser test (DatadogGenericResource)",
+      "options": {
+        [...]
+      },
+      "tags": [
+        "foo:bar"
+      ],
+      "type": "browser",
+      "steps": [
+        [...]
+      ]
+    }
+```
+
+A `DatadogGenericResource` object has two fields:
+* `type`: as one of the [supported resources types](#supported-resources) (e.g. `synthetics_browser_test`), API to create the resource in the Datadog platform.
```

Comment:
Should `type` just be a string from the supported resource types below? Maybe remove the "API to create..." section for clarity?